app/models/organisation.rb

```diff
@@ -53,6 +53,16 @@ class Organisation < ApplicationRecord
 
   scope :only_faith_schools, -> { where.not("gias_data ->> 'ReligiousCharacter (name)' IN (?)", NON_FAITH_RELIGIOUS_CHARACTER_TYPES) }
 
+  scope :with_live_vacancies, lambda {
+    organisations = Vacancy.live.map(&:organisations).flatten.uniq
+    local_authorities_extra_schools = Rails.configuration.local_authorities_extra_schools.to_h.transform_keys(&:to_s)
+
+    local_authorities = local_authorities_extra_schools.select { |_la_code, school_urns| school_urns.any? { |urn| organisations.map(&:urn).include? urn.to_s } }
+
+    all_organisations = organisations.map { |org| [org] + org.school_groups }.flatten + SchoolGroup.where(local_authority_code: local_authorities.keys)
+    where(id: all_organisations.map(&:id))
+  }
+
```

Comment:
I found this took quite a lot of work to understand but also struggling to simplify i guess it's just a tricky one!